moztelemetry/heka/message_parser.py

```diff
@@ -205,9 +205,9 @@ def read_one_record(input_stream, raw=False, verbose=False, strict=False, try_sn
     header.ParseFromString(header_raw)
     unit_separator = input_stream.read(1)
     total_bytes += 1
-    if ord(unit_separator[0]) != 0x1f:
+    if ord(unit_separator) != 0x1f:
```

Comment:
Did this fail with the `[0]` left in? It seems to me it should behave the same way with or without `[0]` and with `StringIO` or `BytesIO`, but I wanted to check and see.

Reply:
in python 3 if you index bytes, you get a number instead of bytes, where in python 2.7 you get a string. but it's already length 1 because of `input_stream.read(1)`, so the `[0]` was unnecessary.

Reply:
Interesting, thanks!